src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageRouteModelProvider.cs

```diff
@@ -97,75 +115,71 @@ private void CreateModels(IList<PageRouteModel> results)
                 {
-                    model = GetAreaPageRouteModel(areaRootDirectory, viewDescriptor);
+                    TryAddAreaRouteModel(context, areaRootDirectory, viewDescriptor);
                 }
                 else if (viewDescriptor.RelativePath.StartsWith(rootDirectory, StringComparison.OrdinalIgnoreCase))
                 {
-                    model = GetPageRouteModel(rootDirectory, viewDescriptor);
-                }
-
-                if (model != null)
-                {
-                    results.Add(model);
+                    AddPageRouteModel(context, rootDirectory, viewDescriptor);
                 }
             }
         }
 
-        private PageRouteModel GetPageRouteModel(string rootDirectory, CompiledViewDescriptor viewDescriptor)
+        private void AddPageRouteModel(
+            PageRouteModelProviderContext context,
+            string rootDirectory,
+            CompiledViewDescriptor viewDescriptor)
         {
             var viewEnginePath = GetRootTrimmedPath(rootDirectory, viewDescriptor.RelativePath);
             if (viewEnginePath.EndsWith(RazorViewEngine.ViewExtension, StringComparison.OrdinalIgnoreCase))
             {
                 viewEnginePath = viewEnginePath.Substring(0, viewEnginePath.Length - RazorViewEngine.ViewExtension.Length);
             }
 
-            var model = new PageRouteModel(viewDescriptor.RelativePath, viewEnginePath);
+            var originalViewEnginePath = viewEnginePath;
             var pageAttribute = (RazorPageAttribute)viewDescriptor.ViewAttribute;
-            PageSelectorModel.PopulateDefaults(model, viewEnginePath, pageAttribute.RouteTemplate);
-            return model;
-        }
-
-        private PageRouteModel GetAreaPageRouteModel(string areaRootDirectory, CompiledViewDescriptor viewDescriptor)
-        {
-            var rootTrimmedPath = GetRootTrimmedPath(areaRootDirectory, viewDescriptor.RelativePath);
 
-            if (PageSelectorModel.TryParseAreaPath(_pagesOptions, rootTrimmedPath, _logger, out var result))
+            var isFallbackRoute = FallbackRazorPagesConventions.TryGetSupersedingPath(viewEnginePath, out var supersedingPath);
+            if (isFallbackRoute)
             {
-                var model = new PageRouteModel(viewDescriptor.RelativePath, result.viewEnginePath)
-                {
-                    RouteValues = { ["area"] = result.areaName },
-                };
-                var pageAttribute = (RazorPageAttribute)viewDescriptor.ViewAttribute;
-                PageSelectorModel.PopulateDefaults(model, result.pageRoute, pageAttribute.RouteTemplate);
-
-                return model;
+                viewEnginePath = supersedingPath;
             }
 
-            // We were unable to parse the path to match the format we expect /Areas/AreaName/Pages/PagePath.cshtml
-            return null;
+            var routeModel = new PageRouteModel(viewDescriptor.RelativePath, viewEnginePath);
```

Comment:
Comments with examples would be good... what are the important properties with and without fallback.